CodeU_Assignement_1/src/codeu_assignement_1/SinglyLinkedList.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package codeu_assignement_1;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ *
+ * @author Karine
+ * 
+ * Bounds, index and general interface is as in Java LinkedList JavaDoc
+ * https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html
+ * 
+ */
+public class SinglyLinkedList <E> {
+    /*  Inner pojo class Node in the list  */
+    public class Node <T>
+    {
+        protected T data;
+        protected Node next;

```

Comment:
I think you misunderstand. The Node variables you've declared are not using the generic parameter, so they're not actually using the generics type system. Unfortunately, the markdown system removed the &lt; and &gt; I wrote. It should have read "So, in this case, it would be Node&lt;T&gt; rather than Node."